Lib/WebHandler.py

```diff
@@ -166,39 +172,49 @@ def __processCredentials(self):
 
     # OIDC auth method
     def oAuth2():
-      if self.get_secure_cookie("AccessToken"):
-        access_token = self.get_secure_cookie("AccessToken")
-        url = Conf.getCSValue("TypeAuths/%s/authority" % typeAuth) + '/userinfo'
-        heads = {'Authorization': 'Bearer ' + access_token, 'Content-Type': 'application/json'}
-        if 'error' in requests.get(url, headers=heads, verify=False).json():
-          self.log.error('OIDC request error: %s' % requests.get(url, headers=heads, verify=False).json()['error'])
-          return
-        ID = requests.get(url, headers=heads, verify=False).json()['sub']
-        result = getUsernameForID(ID)
-        if result['OK']:
-          self.__credDict['username'] = result['Value']
-        result = getDNForUsername(self.__credDict['username'])
-        if result['OK']:
-          self.__credDict['validDN'] = True
-          self.__credDict['DN'] = result['Value'][0]
-        result = getCAForUsername(self.__credDict['username'])
+      if self.get_secure_cookie("StateAuth"):
+        state = self.get_secure_cookie("StateAuth")
+        result = oauth.getUsrnameForState(state) if oauth else S_ERROR('OAuthDIRAC extantion is not enable')
+        if not result['OK']:
+          return result
+        self.set_secure_cookie("StateAuth", result['Value']['state'])
```

Comment:
probably better: result['Value'].get('state')

Reply:
Added condition that must check dict